homework-9/script-1.js

```diff
@@ -0,0 +1,187 @@
+    this.countryPerson = countryPerson;
+    this.listCountryForTravelling = listCountryForTravelling;
+
+}
+const acquaintance = function () {
+
+    console.log(`Hello ${this.namePerson}. Your information: your age is ${this.agePerson}, your sex is ${this.sexPerson}, your nationality is ${this.nationalityPerson}, your country is ${this.countryPerson}`);
+}
+
+
+const wakeUp = function () {
+
+    console.log(`Hello ${this.namePerson}. It's time to wake up!`);
+}
+
+
+const goToBed = function () {
+
+    console.log(`Hello ${this.namePerson}. It's time to go to bed!`);
+}
+
+const travellInCountries = function () {
+
+    console.log(`${this.namePerson} want to travell in such countries as ${this.listCountryForTravelling}`);
+}
+
+const checkInfo = function () {
+    if (Number.isInteger(this.namePerson) || this.namePerson === undefined || this.namePerson === null ||
+        !Number.isInteger(this.agePerson) || this.agePerson < 0 ||
+        Number.isInteger(this.sexPerson) || this.sexPerson === undefined || this.sexPerson === null ||
+        Number.isInteger(this.nationalityPerson) || this.nationalityPerson === undefined || this.nationalityPerson === null ||
+        Number.isInteger(this.countryPerson) || this.countryPerson === undefined || this.countryPerson === null ||
+        Number.isInteger(this.listCountryForTravelling) || this.listCountryForTravelling === undefined || this.listCountryForTravelling === null) {
+        console.log('Sorry, something went wrong');
+    } else {
+        acquaintance();
+        goToBed();
+        wakeUp();
+        travellInCountries();
+    }
+}
+const person2 = new Person('Olga', 29, 'female', 'ukrainian', 'Ukraine', ['Austria', 'Poland', 'Germany']);
+checkInfo.apply(person2);
+
+//Створити власну реалізацію методу .bind (підказка: в кінці треба помістити цю власну 
+//функцію у прототип - Function.prototype.myOwnBind = function () { [тут_код_кастомного_bind] }
+function Dog(nameDog, ageDog, sexDog) {
+    this.nameDog = nameDog;
+    this.ageDog = ageDog;
+    this.sexDog = sexDog;
+}
+const myOwnBind = function () {
+    if (Number.isInteger(this.nameDog) || this.nameDog === undefined || this.nameDog === null ||
+        !Number.isInteger(this.ageDog) || this.ageDog < 0 ||
+        Number.isInteger(this.sexDog) || this.sexDog === undefined || this.sexDog === null) {
+        console.log('Sorry, something went wrong');
+    } else {
+        console.log(`Hello dog ${this.nameDog}. Your information: your age is ${this.ageDog}, your sex is ${this.sexDog}`);
+    }
+
+}
+const dog1 = new Dog('Rex', 5, 'male');
+myOwnBind.bind(dog1)();
+
+//Cтворити функцію, котра приймає 2 параметри - об'єкти. Функція повинна перевіряти чи ці 2 обʼєкти абсолютно ідентичні та повертає результат у зрозумілому форматі
+const beverages1 = {
+    name: 'majito',
+    kind: 'non-alcoholic'
+}
+const beverages2 = {
+    name: 'blue lagoon',
+    kind: 'alcoholic'
+}
+
+const checkObj = function (obj1, obj2) {
```

Comment:
It does not work if we have nested objects.
Also, draw your attention to the qty of results appearing in the console. It's better to notify that objects are identical as a result of function calculation but not as a result of each comparison
<img width="540" alt="image" src="https://user-images.githubusercontent.com/84010501/194805551-1a211204-d01f-4600-b682-6c3147e64755.png">